cvat/apps/engine/signals.py

```diff
@@ -3,21 +3,17 @@
 # SPDX-License-Identifier: MIT
 import shutil
 
+from django.contrib.auth.models import User
 from django.db.models.signals import post_delete, post_save
 from django.dispatch import receiver
-from django.contrib.auth.models import User
 
-from .models import (
-    Data,
-    Job,
-    StatusChoice,
-    Task,
-    Profile,
-)
+from .models import (CloudStorage, Data, Job, Profile, Project,
+    StatusChoice, Task)
 
 
-@receiver(post_save, sender=Job, dispatch_uid="update_task_status")
-def update_task_status(instance, **kwargs):
+@receiver(post_save, sender=Job,
```

Comment:
Directories are always different. They have ID of objects from DB inside. It is not possible.

Reply:
As I remember, owning entity id (e.g. task, project) and data ids are different. There are tests in the [PR](https://github.com/opencv/cvat/pull/5083/files#diff-890920c4573bb488ff08dd5351c6d43e2a22bfbcccd5ee61fa53e9f37657bb46R56-R59) that found a case of collision (in a purely testing scenario yet). Can it become a problem in this solution?

Reply:
@zhiltsov-max , could you please give me more information about the problem you mean? Basically the code below removes directories:
- For Jobs: /home/django/data/jobs/<id>
- For Tasks: /home/django/data/tasks/<id>
- For Projects: /home/django/data/projects/<id>
- For Data: /home/django/data/data/<id>

Reply:
From the linked fragment of code:
```
# The parent class clears DB changes, and it can lead to under-cleaned task data,
# which can affect other tests.
# This situation is not expected to happen on a real server, because
# cache keys include Data object ids, which cannot be reused or freed.
```
Thus, if:
1. A new task is created
2. Task data is uploaded
3. The task removed
4. A new task is created
5. The second task data is uploaded, the files have overlapping names and the new data directory id is the same as the first task had, and the cache is used

then the second task can look to the same cache keys as the first task had and receive matching frames from the first task data, instead of the new data. AFAIK, this should not happen on real servers, because data ids always grow.

Reply:
Perhaps we can guard against this by checking that when we create a data directory, it does not already exist?